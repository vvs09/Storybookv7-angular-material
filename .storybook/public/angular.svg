<svg width="188" height="200" viewBox="0 0 188 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M94 123.4H69.3L57.6 152.6H35.9H35.8L94 22.1V22.2V0L0.899994 33.2L15.1 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M77 105.4H94V64.5L77 105.4Z" fill="#808080"/>
<g opacity="0.9">
<path d="M94 123.4H118.7L130.4 152.6H152.1H152.2L94 22.1V22.2V0L187.1 33.2L172.9 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M111 105.4H94V64.5L111 105.4Z" fill="#808080"/>
</g>
</svg>
