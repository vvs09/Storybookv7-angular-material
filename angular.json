{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "storybook": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "sass"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/storybook",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "storybook:build:production"
            },
            "development": {
              "buildTarget": "storybook:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "storybook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "port": 6006,
            "enableProdMode":true
          }
        },
        "build-storybook": {
          "builder": "@storybook/angular:build-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "outputDir": "storybook-static"
          }
        }
      }
    }
  }
}