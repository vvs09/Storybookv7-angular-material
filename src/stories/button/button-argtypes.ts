export const buttonArgTypes= {
    color: {
      control: 'radio',
      options: ['accent', 'primary', 'warn'],
      description: 'Theme color palette for the component.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'null' },
        type: {
          summary: 'string',
        },
      },
    },
    disabled: {
      control: 'boolean',
      description:
        'Whether the checkbox is disabled. This fully overrides the implementation provided by mixinDisabled, but the mixin is still required because mixinTabIndex requires it.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
    disableRipple: {
      control: 'boolean',
      description: 'Whether ripples are disabled.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
    buttonType: {
      control: 'select',
      options: [
        'mat-button',
        'mat-raised-button',
        'mat-stroked-button',
        'mat-flat-button',
      ],
      description: 'Type of buton.',
      table: {
        defaultValue: { summary: 'mat-button' },
        type: {
          summary: 'string',
        },
      },
    },
    label: {
      control: 'text',
      description: 'Button label.',
      table: {
        defaultValue: { summary: 'button' },
        type: {
          summary: 'string',
        },
      },
    },
    ariaDisabled: {
      control: 'boolean',
      description: 'aria-disabled value of the button.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'undefined' },
        type: {
          summary: 'boolean | undefined',
        },
      },
    },
    disabledInteractive: {
      control: 'boolean',
      description:
        'Natively disabled buttons prevent focus and any pointer events from reaching the button. In some scenarios this might not be desirable, because it can prevent users from finding out why the button is disabled (e.g. via tooltip). Enabling this input will change the button so that it is styled to be disabled and will be marked as aria-disabled, but it will allow the button to receive events and focus. Note that by enabling this, you need to set the tabindex yourself if the button isnt meant to be tabbable and you have to prevent the button action (e.g. form submissions).',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
  }