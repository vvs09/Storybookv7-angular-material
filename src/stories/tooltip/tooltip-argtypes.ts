export const tooltipArgTypes = {
    disabled: {
        control: 'boolean',
        description: 'Disables the display of the tooltip.',
        table: {
          category: 'Behavior',
          defaultValue: { summary: 'false' },
          type: { summary: 'boolean' },
        },
      },
      hideDelay: {
        control: 'number',
        description: 'The default delay in ms before hiding the tooltip after hide is called.',
        table: {
          category: 'Timing',
          defaultValue: { summary: '0' },
          type: { summary: 'number' },
        },
      },
      message: {
        control: 'text',
        description: 'The message to be displayed in the tooltip.',
        table: {
          category: 'Content',
          defaultValue: { summary: '' },
          type: { summary: 'string' },
        },
      },
      position: {
        control: 'select',
        description: 'Allows the user to define the position of the tooltip relative to the parent element.',
        options: ['above', 'below', 'left', 'right', 'before', 'after'],
        table: {
          category: 'Positioning',
          defaultValue: { summary: 'below' },
          type: { summary: 'TooltipPosition' },
        },
      },
      positionAtOrigin: {
        control: 'boolean',
        description: 'Whether tooltip should be relative to the click or touch origin instead of outside the element bounding box.',
        table: {
          category: 'Positioning',
          defaultValue: { summary: 'false' },
          type: { summary: 'boolean' },
        },
      },
      showDelay: {
        control: 'number',
        description: 'The default delay in ms before showing the tooltip after show is called.',
        table: {
          category: 'Timing',
          defaultValue: { summary: '0' },
          type: { summary: 'number' },
        },
      },
      tooltipClass: {
        control: 'object',
        description: 'Classes to be passed to the tooltip. Supports the same syntax as ngClass.',
        table: {
          category: 'Styling',
          type: { summary: 'any' },
        },
      },
      touchGestures: {
        control: 'select',
        description: 'How touch gestures should be handled by the tooltip.',
        options: ['auto', 'on', 'off'],
        table: {
          category: 'Behavior',
          defaultValue: { summary: 'auto' },
          type: { summary: 'TooltipTouchGestures' },
        },
      }
  };
  