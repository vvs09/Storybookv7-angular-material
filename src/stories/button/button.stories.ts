import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatButtonModule, MatButton } from '@angular/material/button';
import { buttonArgTypes } from './button-argtypes';

const meta: Meta<MatButton & { label: string; buttonType: string }> = {
  title: 'Components/Button',
  component: MatButton,
  tags: ['autodocs'],
  argTypes: {
    ...buttonArgTypes,
  },
  decorators: [
    moduleMetadata({
      imports: [MatButtonModule],
    }),
  ],
  render: (args) => ({
    props: args,
    template: `
    <button  *ngIf="buttonType === 'mat-button'" mat-button  ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-raised-button' " mat-raised-button ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-stroked-button' " mat-stroked-button ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-flat-button' "  mat-flat-button ${argsToTemplate(
      args
    )} >{{label}}</button>
\    `,
  }),
};

export default meta;
type Story = StoryObj<MatButton & { label: string; buttonType: string }>;

export const accent: Story = {
  args: {
    color: 'accent',
    disabled: false,
    ariaDisabled: false,
    disabledInteractive: false,
    disableRipple: false,
    label: 'Button',
    buttonType: 'mat-button',
  },
};

export const Primary: Story = {
  args: {
    ...accent.args,
    color: 'primary',
  },
};

export const Warn: Story = {
  args: {
    ...accent.args,
    color: 'warn',
  },
};

export const Disabled: Story = {
  args: {
    ...accent.args,
    disabled: true,
  },
};

export const matRaisedButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-raised-button',
  },
};

export const matRaisedButtonPrimary: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    color: 'primary',
  },
};

export const matRaisedButtonWarn: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    color: 'warn',
  },
};

export const MatRaisedButtonDisabled: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    disabled: true,
  },
};

export const MatStrokedButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-stroked-button',
  },
};

export const MatStrokedButtonPrimary: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    color: 'primary',
  },
};

export const MatStrokedButtonWarn: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    color: 'warn',
  },
};

export const MatStrokedButtonDisabled: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    disabled: true,
  },
};

export const MatFlatButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-flat-button',
  },
};

export const MatFlatButtonPrimary: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    color: 'primary',
  },
};

export const MatFlatButtonWarn: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    color: 'warn',
  },
};

export const MatFlatButtonDisabled: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    disabled: true,
  },
};
