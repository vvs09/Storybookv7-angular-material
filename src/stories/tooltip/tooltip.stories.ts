import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatTooltipModule, MatTooltip } from '@angular/material/tooltip';

import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatButtonModule } from '@angular/material/button';
import { tooltipArgTypes } from './tooltip-argtypes';

const meta: Meta<MatTooltip> = {
  title: 'Components/Tooltip',
  component: MatTooltip,
  tags: ['autodocs'],
  argTypes: tooltipArgTypes,
  decorators: [
    moduleMetadata({
      imports: [
        MatTooltipModule,
        MatButtonModule,
        ReactiveFormsModule,
        FormsModule,
      ],
    }),
  ],
  render: (args) => ({
    props: args,
    styles: [
      `
      .example-button {
        display: block;
        margin: 80px auto 400px;
      }
      
      .example-container {
        height: 200px;
        overflow: auto;
        border: 1px solid #ccc;
      }
        `,
    ],
    template: `
    <div class="example-container" cdkScrollable>
    <button mat-raised-button
    [matTooltip]
    ${argsToTemplate(args)}
    class="example-button">
     Tooltip
    </button>
    </div>
   `,
  }),
};

export default meta;
type Story = StoryObj<MatTooltip>;

export const Basic: Story = {
  args: {
    message: 'Tooltip',
    disabled: false,
    touchGestures: 'auto',
    hideDelay: 0,
    showDelay: 0,
    position: 'before',
    positionAtOrigin: false,
  },
};

export const hideDelay: Story = {
  args: {
    ...Basic.args,
    hideDelay: 1000,
  },
};

export const showDelay: Story = {
  args: {
    ...Basic.args,
    showDelay: 1000,
  },
};

export const PositionAbove: Story = {
  args: {
    ...Basic.args,
    position: 'above',
  },
};

export const PositionAfter: Story = {
  args: {
    ...Basic.args,
    position: 'after',
  },
};

export const PositionBefore: Story = {
  args: {
    ...Basic.args,
    position: 'before',
  },
};

export const PositionLeft: Story = {
  args: {
    ...Basic.args,
    position: 'left',
  },
};

export const PositionRight: Story = {
  args: {
    ...Basic.args,
    position: 'right',
  },
};
