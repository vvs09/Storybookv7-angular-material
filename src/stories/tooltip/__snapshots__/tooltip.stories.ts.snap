// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Tooltip Basic smoke-test 1`] = `
<storybook-root _nghost-ng-c3197468757
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c3197468757
       cdkscrollable
       class="example-container"
  >
    <button _ngcontent-ng-c3197468757
            mat-raised-button
            class="mat-mdc-tooltip-trigger example-button mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
            mat-ripple-loader-uninitialized
            mat-ripple-loader-class-name="mat-mdc-button-ripple"
            ng-reflect-disabled="false"
            aria-describedby="cdk-describedby-message-ng-1-2"
            cdk-describedby-host="ng-1"
    >
      <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
      </span>
      <span class="mdc-button__label">
        Tooltip
      </span>
      <span class="mat-mdc-focus-indicator">
      </span>
      <span class="mat-mdc-button-touch-target">
      </span>
    </button>
  </div>
</storybook-root>
`;

exports[`Components/Tooltip PositionAbove smoke-test 1`] = `
<storybook-root _nghost-ng-c3197468757
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c3197468757
       cdkscrollable
       class="example-container"
  >
    <button _ngcontent-ng-c3197468757
            mat-raised-button
            class="mat-mdc-tooltip-trigger example-button mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
            mat-ripple-loader-uninitialized
            mat-ripple-loader-class-name="mat-mdc-button-ripple"
            ng-reflect-disabled="false"
            aria-describedby="cdk-describedby-message-ng-10-11"
            cdk-describedby-host="ng-10"
    >
      <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
      </span>
      <span class="mdc-button__label">
        Tooltip
      </span>
      <span class="mat-mdc-focus-indicator">
      </span>
      <span class="mat-mdc-button-touch-target">
      </span>
    </button>
  </div>
</storybook-root>
`;

exports[`Components/Tooltip PositionAfter smoke-test 1`] = `
<storybook-root _nghost-ng-c3197468757
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c3197468757
       cdkscrollable
       class="example-container"
  >
    <button _ngcontent-ng-c3197468757
            mat-raised-button
            class="mat-mdc-tooltip-trigger example-button mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
            mat-ripple-loader-uninitialized
            mat-ripple-loader-class-name="mat-mdc-button-ripple"
            ng-reflect-disabled="false"
            aria-describedby="cdk-describedby-message-ng-13-14"
            cdk-describedby-host="ng-13"
    >
      <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
      </span>
      <span class="mdc-button__label">
        Tooltip
      </span>
      <span class="mat-mdc-focus-indicator">
      </span>
      <span class="mat-mdc-button-touch-target">
      </span>
    </button>
  </div>
</storybook-root>
`;

exports[`Components/Tooltip PositionBefore smoke-test 1`] = `
<storybook-root _nghost-ng-c3197468757
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c3197468757
       cdkscrollable
       class="example-container"
  >
    <button _ngcontent-ng-c3197468757
            mat-raised-button
            class="mat-mdc-tooltip-trigger example-button mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
            mat-ripple-loader-uninitialized
            mat-ripple-loader-class-name="mat-mdc-button-ripple"
            ng-reflect-disabled="false"
            aria-describedby="cdk-describedby-message-ng-16-17"
            cdk-describedby-host="ng-16"
    >
      <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
      </span>
      <span class="mdc-button__label">
        Tooltip
      </span>
      <span class="mat-mdc-focus-indicator">
      </span>
      <span class="mat-mdc-button-touch-target">
      </span>
    </button>
  </div>
</storybook-root>
`;

exports[`Components/Tooltip PositionLeft smoke-test 1`] = `
<storybook-root _nghost-ng-c3197468757
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c3197468757
       cdkscrollable
       class="example-container"
  >
    <button _ngcontent-ng-c3197468757
            mat-raised-button
            class="mat-mdc-tooltip-trigger example-button mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
            mat-ripple-loader-uninitialized
            mat-ripple-loader-class-name="mat-mdc-button-ripple"
            ng-reflect-disabled="false"
            aria-describedby="cdk-describedby-message-ng-19-20"
            cdk-describedby-host="ng-19"
    >
      <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
      </span>
      <span class="mdc-button__label">
        Tooltip
      </span>
      <span class="mat-mdc-focus-indicator">
      </span>
      <span class="mat-mdc-button-touch-target">
      </span>
    </button>
  </div>
</storybook-root>
`;

exports[`Components/Tooltip PositionRight smoke-test 1`] = `
<storybook-root _nghost-ng-c3197468757
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c3197468757
       cdkscrollable
       class="example-container"
  >
    <button _ngcontent-ng-c3197468757
            mat-raised-button
            class="mat-mdc-tooltip-trigger example-button mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
            mat-ripple-loader-uninitialized
            mat-ripple-loader-class-name="mat-mdc-button-ripple"
            ng-reflect-disabled="false"
            aria-describedby="cdk-describedby-message-ng-22-23"
            cdk-describedby-host="ng-22"
    >
      <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
      </span>
      <span class="mdc-button__label">
        Tooltip
      </span>
      <span class="mat-mdc-focus-indicator">
      </span>
      <span class="mat-mdc-button-touch-target">
      </span>
    </button>
  </div>
</storybook-root>
`;

exports[`Components/Tooltip hideDelay smoke-test 1`] = `
<storybook-root _nghost-ng-c3197468757
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c3197468757
       cdkscrollable
       class="example-container"
  >
    <button _ngcontent-ng-c3197468757
            mat-raised-button
            class="mat-mdc-tooltip-trigger example-button mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
            mat-ripple-loader-uninitialized
            mat-ripple-loader-class-name="mat-mdc-button-ripple"
            ng-reflect-disabled="false"
            aria-describedby="cdk-describedby-message-ng-4-5"
            cdk-describedby-host="ng-4"
    >
      <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
      </span>
      <span class="mdc-button__label">
        Tooltip
      </span>
      <span class="mat-mdc-focus-indicator">
      </span>
      <span class="mat-mdc-button-touch-target">
      </span>
    </button>
  </div>
</storybook-root>
`;

exports[`Components/Tooltip showDelay smoke-test 1`] = `
<storybook-root _nghost-ng-c3197468757
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c3197468757
       cdkscrollable
       class="example-container"
  >
    <button _ngcontent-ng-c3197468757
            mat-raised-button
            class="mat-mdc-tooltip-trigger example-button mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
            mat-ripple-loader-uninitialized
            mat-ripple-loader-class-name="mat-mdc-button-ripple"
            ng-reflect-disabled="false"
            aria-describedby="cdk-describedby-message-ng-7-8"
            cdk-describedby-host="ng-7"
    >
      <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
      </span>
      <span class="mdc-button__label">
        Tooltip
      </span>
      <span class="mat-mdc-focus-indicator">
      </span>
      <span class="mat-mdc-button-touch-target">
      </span>
    </button>
  </div>
</storybook-root>
`;
